gr-digital/lib/costas_loop_cc_impl.cc

```diff
@@ -31,9 +31,28 @@
 #include <gnuradio/sincos.h>
 #include <boost/format.hpp>
 
+#define CL_TWO_PI 6.28318530717958647692
+#define CL_ONE_OVER_2PI 0.15915494309189533577
+#define CL_MINUS_TWO_PI -6.28318530717958647692
+
+#if !defined(__FMA__) && defined(__AVX2__)
+#define __FMA__ 1
+#endif
+
+#if defined(FP_FAST_FMA)
+#define __FMA__ 1
+#endif
+
 namespace gr {
 namespace digital {
 
+struct ComplexStruct {
```

Comment:
Is there a reason to use this instead of gr_complex?

Reply:
Reaching for every ounce of performance, each real() and imag() call in the loop with gr_complex was causing function jumps.  the ComplexStruct overlays a struct over the same memory to access the data without the jumps.  In general I wouldn't worry too much about it, but when you wrap it up in a work for loop, the gain adds up.

Reply:
I can not reproduce this with current master, compiling with GCC 9.2 (-O2). I see no functions calls, i.e. ``complex::real()/imag()`` is implemented (as expected) efficiently.

Reply:
I'd guess that with `-O2` optimization enabled using this or `std::complex` will result in roughly the same code and code speed, since the calls to `real()` and `imag()` should be optimized out by the compiler. My preference would be to use `gr_complex` and related class calls here.

Reply:
Echo this: I also prefer gr_complex. I would be very surprised that a local hack will beat the STL performance, especially with other optimization levels.